src/render/python/integrator_v.cpp

```diff
@@ -247,28 +247,28 @@ MI_VARIANT class PyADIntegrator : public CppADIntegrator<Float, Spectrum> {
                                      Sampler *sampler,
                                      const RayDifferential3f &ray,
                                      const Medium * /* unused */,
-                                     Float * /* unused */,
+                                     Float * aovs,
                                      Mask active) const override {
         py::gil_scoped_acquire gil;
 
         py::function sample_override = py::get_override(this, "sample");
         if (sample_override) {
-            using PyReturn = std::tuple<Spectrum, Mask, py::object>;
-
+            using PyReturn = std::tuple<Spectrum, Mask, std::vector<Float>, py::object>;
             py::kwargs kwargs = py::dict(
                 "mode"_a=drjit::ADMode::Primal,
                 "scene"_a=scene,
                 "sampler"_a=sampler,
                 "ray"_a=ray,
                 "depth"_a=0,
                 "δL"_a=py::none(),
+                "δaovs"_a=py::none(),
                 "state_in"_a=py::none(),
                 "active"_a=active
             );
             // Third output is the ADIntegrator's state, which we ignore here
-            auto [spec, mask, _] =
+            auto [spec, mask, aovs_, _] =
                 sample_override(**kwargs).template cast<PyReturn>();
-
+            std::copy(aovs_.begin(), aovs_.end(), aovs);
```

Comment:
Does this actually work? I suppose at this point `aovs_` is already converted to a `std::vector<Float>`?

In my case I was casting the variables explicitly:

```c++
    // Convert AOVs from python object to float array if needed
    if (py::isinstance<py::list>(outputs)) {
        auto list = outputs.template cast<py::list>();
        for (size_t i = 0; i < list.size(); ++i)
            aovs[i] = list[i].template cast<Float>();
    }
 ```

Reply:
Yes, I think so. I followed the already existing code from the PySamplingIntegrator. We are explicitly casting the return type to an std::vector. I think the code you wrote is marginally more efficient, as it does not require the temporary std::vector, but I would favor readability & simplicity here.